app/src/main/java/de/westnordost/streetcomplete/overlays/shops/ShopsOverlay.kt

```diff
@@ -41,10 +42,7 @@ class ShopsOverlay(private val featureDictionaryFuture: FutureTask<FeatureDictio
         mapData
             .filter(IS_SHOP_OR_DISUSED_SHOP_EXPRESSION)
             .map { element ->
-                val feature = featureDictionaryFuture.get()
-                    .byTags(element.tags)
-                    .isSuggestion(false) // no suggestions because we just want the icon
-                    .find().firstOrNull()
+                val feature = featureDictionaryFuture.get().getFeature(element.tags)
```

Comment:
The feature dictionary here is used the same way as in the quests. So, instead of the future, just pass the function in the constructor of ShopsOverlay

Reply:
Done in [`2bd3207`](https://github.com/streetcomplete/StreetComplete/pull/5273/commits/2bd32072a828d5c47b630e48a53b814e343ec3f5).